### PR TITLE
Ntrnl 342 ntrnl 343 ntrnl 345 add logic and model to save user data to session for add member repo and team html page

### PR DESCRIPTION
**Pull request by Mouhajer-CO**
### JIRA link

Resolves: [NTRNL-342](https://technologyprogramme.atlassian.net/browse/NTRNL-342), [NTRNL-343](https://technologyprogramme.atlassian.net/browse/NTRNL-343) and [NTRNL-345](https://technologyprogramme.atlassian.net/browse/NTRNL-345)

### Description

- Add security-check target on Makefile 
- Add session storage to `add-member` page and update unit tests
- Add session storage to `add-repo` page and update unit tests
- Add session storage to `add-team` page and update unit tests

### Work checklist

- [x] Tests added where applicable
- [x] No vulnerability added

[NTRNL-342]: https://technologyprogramme.atlassian.net/browse/NTRNL-342?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[NTRNL-343]: https://technologyprogramme.atlassian.net/browse/NTRNL-343?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[NTRNL-345]: https://technologyprogramme.atlassian.net/browse/NTRNL-345?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/controller/add-member.controller.ts [hunk 1]
```diff
@@ -1,22 +1,38 @@
...
-    const emailAddress = req.body.email_address;
-    const contractType = req.body.contract_type;
+export const post = (req: Request, res: Response, next: NextFunction ) => {
+    try {
+        const firstName = req.body.first_name;
+        const lastName = req.body.last_name;
+        const gitHubHandle = req.body.github_handle;
+        const emailAddress = req.body.email_address;
+        const contractType = req.body.contract_type;
+        const contract_end_date = req.body.contract_end_date;
```
Comment (harley-harris) [inline src/controller/add-member.controller.ts:20]: camelCase for consistency

File: src/controller/add-repo.controller.ts [hunk 2]
```diff
@@ -1,19 +1,30 @@
...
+export const post = (req: Request, res: Response, next: NextFunction ) => {
+    try {
+        const repoName = req.body.repo_name;
+        const visibility = req.body.visibility;
 
-    // validation middleware and data assignment to be implemented
+        log.info(`Repository Name: ${repoName}, Visibility: ${visibility}`);
 
-    log.info(`Repository Name: ${repoName}, Visibility: ${visibility}`);
+        setSessionData(req.session, {
```
[the remark this answers is not in the document]
Comment (Mouhajer-CO) [reply src/controller/add-repo.controller.ts:20]: Agreed, we could eventually add more specific method like getSessionByKey, setSessionByKey or updateSessionData and more as utils.

[the remark this answers is not in the document]
Comment (Mouhajer-CO) [reply src/controller/add-repo.controller.ts:20]: We will discuss it during the standup and add a related card/story in the backlog

**Review by harley-harris — APPROVED** [review]
Great work again @Mouhajer-CO. I tested data persistance by logging the session object in each of the controllers `console.log(getSessionData(req.session));` 

Screenshot 2024-03-01 at 12 07 49

🚀

**Pull request merged**